### PR TITLE
[Deepseek V3.2] Use torch.compile to speed up torch.cat in nsa

### PR DESCRIPTION
**Pull request by hlu1**
## Motivation

Use torch.compile to speed up torch.cat used in prefill.

## Modifications

Replace the cat ops for k in `forward_extend` in the nsa attention with the torch.compiled version.

## Accuracy Tests
```
gpqa

Repeat: 8, mean: 0.803

Scores: ['0.813', '0.803', '0.798', '0.783', '0.823', '0.803', '0.788', '0.813']

gsm8k with 20-shots

Accuracy: 0.956
Invalid: 0.000
Latency: 25.842 s
Output throughput: 5056.162 token/s
```

## Benchmarking and Profiling

```
before:

============ Serving Benchmark Result ============
Backend:                                 sglang
Traffic request rate:                    inf
Max request concurrency:                 not set
Successful requests:                     8
Benchmark duration (s):                  20.17
Total input tokens:                      32768
Total input text tokens:                 32768
Total input vision tokens:               0
Total generated tokens:                  128
Total generated tokens (retokenized):    124
Request throughput (req/s):              0.40
Input token throughput (tok/s):          1624.55
Output token throughput (tok/s):         6.35
Total token throughput (tok/s):          1630.90
Concurrency:                             0.84

after


============ Serving Benchmark Result ============
Backend:                                 sglang
Traffic request rate:                    inf
Max request concurrency:                 not set
Successful requests:                     8
Benchmark duration (s):                  19.92
Total input tokens:                      32768
Total input text tokens:                 32768
Total input vision tokens:               0
Total generated tokens:                  128
Total generated tokens (retokenized):    124
Request throughput (req/s):              0.40
Input token throughput (tok/s):          1645.08
Output token throughput (tok/s):         6.43
Total token throughput (tok/s):          1651.50
Concurrency:                             0.85
```

before
image

after
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by hlu1** [comment]
> @hlu1 Have you tried `_concat_mla_absorb_q_general` , like in this PR https://github.com/sgl-project/sglang/pull/12215/files

Yes, I put the perf comparison results and next steps in the Benchmarking and Profiling section

> https://github.com/sgl-project/sglang/pull/12215 is faster for prefill, but _concat_mla_absorb_q_general only works for prefill. We can use _concat_mla_absorb_q_general for prefill and torch.compile for decode after https://github.com/sgl-project/sglang/pull/12215 is merged.

**Review by Fridge003 — APPROVED** [review]

**Review by Fridge003 — APPROVED** [review]
